src/components/Layout/Layout.stories.tsx

```diff
@@ -150,6 +164,83 @@ function ExamplePageComponent() {
   );
 }
 
+export function FormPageLayout() {
+  const formState = useFormState({
+    config: formConfig,
+    init: { input: { firstName: "John", middleInitial: "C", lastName: "Doe" } },
+  });
+
+  return (
+    <FormPageLayoutComponent
+      pageTitle="Detail Title"
+      submitAction={{ label: "Save", onClick: noop }}
+      cancelAction={{ label: "Cancel", onClick: noop }}
+      tertiaryAction={{ label: "Tertiary Test", onClick: noop }}
+      breadCrumb={[
+        { label: "Breadcrumb A", href: "/breadcrumb-a" },
+        { label: "Breadcrumb B", href: "/breadcrumb-b" },
+      ]}
+      formSections={formSections}
+      formState={formState}
+    />
+  );
+}
+
+const sportsOptions = [
+  { id: "s:1", name: "Basketball" },
+  { id: "s:2", name: "Baseball" },
+];
+
+const colorOptions = [
+  { id: "c:1", name: "Red" },
+  { id: "c:2", name: "Blue" },
+  { id: "c:3", name: "Green" },
+];
+
+const formSections: FormSectionConfig<AuthorInput> = [
+  {
+    title: "Author Overview",
+    icon: "userCircle",
+    rows: [
+      { firstName: boundTextField(), middleInitial: boundTextField(), lastName: boundTextField() },
+      { bio: boundTextAreaField() },
+    ],
+  },
+  {
+    title: "More Details",
+    icon: "openBook",
+    rows: [
+      {
+        favoriteSport: boundSelectField({
+          options: sportsOptions,
+          getOptionLabel: (o) => o.name,
+          getOptionValue: (o) => o.id,
+        }),
+        favoriteColors: boundMultiSelectField({
+          options: colorOptions,
+          getOptionLabel: (o) => o.name,
+          getOptionValue: (o) => o.id,
+        }),
+      },
+      { heightInInches: boundNumberField({ label: "Height (in inches)" }), birthday: boundDateField() },
+      { isAvailable: boundCheckboxField({ label: "Is Retired" }) },
+    ],
+  },
+];
+
+const formConfig: ObjectConfig<AuthorInput> = {
+  firstName: { type: "value", rules: [required] },
+  middleInitial: { type: "value" },
+  lastName: { type: "value", rules: [required] },
+  birthday: { type: "value", rules: [required] },
+  heightInInches: { type: "value" },
+  favoriteSport: { type: "value" },
+  favoriteColors: { type: "value" },
+  bio: { type: "value" },
+  isAvailable: { type: "value" },
+  favoriteShapes: { type: "value" },
+};
+
```

Comment:
I wish we could start assuming `type: "value"`. With the _rare_ exception of some `list` these appear to be 99% of all of our form inputs. Lots of boilerplate.

Reply:
Yep; that's what this was trying to fix:

https://github.com/homebound-team/form-state/blob/main/src/configBuilders.test.ts#L6

In general, the libraries I build often like to have "ASTs of their data", and I start out with "just let the user fill out that AST by hand" (they are "just POJOs" ==> easy in JS right?), but ASTs by their nature are not usually succinct, so adding a "builder DSL" at some point is a reasonable next-step/makes sense.